## Cabeçalho

In [1]:
import pandas as pd
import numpy as np

data = pd.read_csv('Twitch_global_data.csv')

data.rename(columns={
    'year':'ano',
    'Month':'mes',
    'Hours_watched':'horas_assistidas',
    'Avg_viewers':'media_visualizacoes',
    'Peak_viewers':'pico_de_visualizacoes',
    'Avg_channels':'media_canais',
    'Games_streamed':'jogos_transmitidos',
    'Viewer_ratio':'proporcao_de_espectadores',
}, inplace=True)

data.tail()

ano  mes  horas_assistidas  media_visualizacoes  pico_de_visualizacoes  \
80  2022    9        1817119612              2527287                4678056   
81  2022   10        1883320673              2534751                4610799   
82  2022   11        1694262814              2356415                5362892   
83  2022   12        1674092138              2253152                3866814   
84  2023    1        1866962462              2512735                5593050   

     Streams  media_canais  jogos_transmitidos  proporcao_de_espectadores  
80  22278079         90376               41863                      28.37  
81  22867973         90557               42164                      28.35  
82  21744817         88195               41058                      26.84  
83  21753674         84728               41548                      27.00  
84  24429476         93349               43611                      27.44

<b><i>Nota: Dada as circunstâncias do ano 2023, por ter apenas o mês de janeiro como referência, o trataremos como um outlier. O valor não será removido ou alterado, porém, em certos casos, poderá ser ignorado.</i></b>

## Etapa Projeto

### Médias por HA e MV

In [64]:
data.head()

ano  mes  horas_assistidas  media_visualizacoes  pico_de_visualizacoes  \
0  2016    1         480241904               646355                1275257   
1  2016    2         441859897               635769                1308032   
2  2016    3         490669308               660389                1591551   
3  2016    4         377975447               525696                1775120   
4  2016    5         449836631               605432                1438962   

   Streams  media_canais  jogos_transmitidos  proporcao_de_espectadores  
0  7701675         20076               12149                      29.08  
1  7038520         20427               12134                      28.98  
2  7390957         20271               12234                      28.92  
3  6869719         16791               12282                      28.80  
4  7535519         19394               12424                      28.85

In [52]:
maior_horas = data.groupby(['ano', 'mes', 'horas_assistidas', 'media_visualizacoes'])['horas_assistidas'].sum().idxmax()
menor_horas = data.groupby(['ano', 'mes', 'horas_assistidas', 'media_visualizacoes'])['horas_assistidas'].sum().idxmin()
maior_horas, menor_horas

((2021, 5, 2308295297, 3106723), (2016, 9, 362833210, 504635))

<b><i>O maior número de horas assistidas foi de 2308295297 horas com 3106723 médias visualizações no mês 5 do ano 2021. O menor número de horas assistidas foi de 362833210 horas com 504635 médias visualizações no mês 9 do ano 2016.</i></b>

In [51]:
maior_visus = data.groupby(['ano', 'mes', 'horas_assistidas', 'media_visualizacoes'])['media_visualizacoes'].sum().idxmax()
menor_visus = data.groupby(['ano', 'mes', 'horas_assistidas', 'media_visualizacoes'])['media_visualizacoes'].sum().idxmin()

maior_visus, menor_visus

((2021, 4, 2238018281, 3112681), (2016, 9, 362833210, 504635))

<b><i>A maior média de visualizações foi de 3112681 visualizações no mês 4 do ano 2021 com 2238018281 horas assistidas. A menor média de visualizações foi de 504635 visualizações no mês 9 do ano 2016 com 362833210 horas assistidas.</i></b>

In [55]:
print(f'A maior média de horas assistidas por visualizações foi de {maior_horas[2] / maior_horas[-1]:.2f}')
print(f'A menor média de horas assistidas por visualizações foi de {menor_horas[2] / menor_horas[-1]:.2f}')

A maior média de horas assistidas por visualizações foi de 743.00
A menor média de horas assistidas por visualizações foi de 719.00


<b><i>Pode-se perceber que, no mês em que há os valores máximo e mínimo de horas assistidas, não há necessariamente os valores máximo e mínimos de médias visualizações.</i></b><br>


### Valores médios por mês e ano

0, 11 -> 2016  <br>
12, 23 -> 2017 <br>
24, 35 -> 2018 <br>
36, 47 -> 2019 <br>
48, 59 -> 2020 <br>
60, 71 -> 2021 <br>
72, 83 -> 2022 <br>
84 -> 2023 <br>

<b>Eu quero saber as médias mensais e anuais de horas assistidas por número de visualizações</b>

In [20]:
def analisaMedias(horas:list(), visus:list()):
    aux = []
    total = []
    ano = 2016
    for x in range(85):
        if x < 12:
            aux.append(horas[x] / visus[x])
            if x == 11:
                aux.append(sum(horas[0:12]) / sum(visus[0:12]))
                aux.insert(0, ano)
                total.append(aux[:])
                aux.clear()
        elif 24 > x > 11:
            aux.append(horas[x] / visus[x])
            if x == 23:
                aux.append(sum(horas[11:24]) / sum(visus[11:24]))
                aux.insert(0, ano + 1)
                total.append(aux[:])
                aux.clear()
        elif 36 > x > 23:
            aux.append(horas[x] / visus[x])
            if x == 35:
                aux.append(sum(horas[24:36]) / sum(visus[24:36]))
                aux.insert(0, ano + 2)
                total.append(aux[:])
                aux.clear()
        elif 48 > x > 35:
            aux.append(horas[x] / visus[x])
            if x == 47:
                aux.append(sum(horas[36:48]) / sum(visus[36:48]))
                aux.insert(0, ano + 3)
                total.append(aux[:])
                aux.clear()
        elif 60 > x > 47:
            aux.append(horas[x] / visus[x])
            if x == 59:
                aux.append(sum(horas[48:60]) / sum(visus[48:60]))
                aux.insert(0, ano + 4)
                total.append(aux[:])
                aux.clear()         
        elif 72 > x > 59:
            aux.append(horas[x] / visus[x])
            if x == 71:
                aux.append(sum(horas[60:72]) / sum(visus[60:72]))
                aux.insert(0, ano + 5) 
                total.append(aux[:])
                aux.clear()
        elif 84 > x > 71:
            aux.append(horas[x] / visus[x])
            if x == 83:
                aux.append(sum(horas[72:84]) / sum(visus[72:84]))
                aux.insert(0, ano + 6)
                total.append(aux[:])
                aux.clear()       
        else:
            aux.append(horas[x] / visus[x])
            if x == 84:
                aux.append((horas[-1]) / (visus[-1]))
                aux.insert(0, ano + 7)
                total.append(aux[:])
                aux.clear()
    return total

   
        

In [21]:
horas_assistidas = [ dados for dados in data['horas_assistidas']]
media_visualizacoes = [dados for dados in data['media_visualizacoes']]

total_medias = analisaMedias(horas_assistidas, media_visualizacoes)

A variável <b>total_medias</b> corresponde a uma lista com outras 8 listas dentro de si. Cada lista dessas possui 1 determinado ano como valor inicial, 12 ou (1) valores de médias mensais de horas assistidas por número de espectadores e, por fim, um valor de média anual do total de horas assistidas pelo total de espectadores anuais.

In [26]:
maiores = []
minimos = []

for x in total_medias:
    maiores.append(f'O maior valor médio de horas por visus do ano {x[0]} é {max(x[1:-1]):.2f} no mês {x.index(max(x[1:-1]))}')
    minimos.append(f'O menor valor médio de horas por visus do ano {x[0]} é {min(x[1:-1]):.2f} no mês {x.index(min(x[1:-1]))}')

In [88]:
for x in range(8):
    print(f'O valor de média anual de {total_medias[x][0]} é de {total_medias[x][-1]:.2f}')

O valor de média anual de 2016 é de 731.19
O valor de média anual de 2017 é de 729.96
O valor de média anual de 2018 é de 729.54
O valor de média anual de 2019 é de 728.81
O valor de média anual de 2020 é de 731.42
O valor de média anual de 2021 é de 728.77
O valor de média anual de 2022 é de 728.38
O valor de média anual de 2023 é de 743.00


In [89]:
maiores, minimos

(['O maior valor médio de horas por visus do ano 2016 é 743.00 no mês 8',
  'O maior valor médio de horas por visus do ano 2017 é 743.00 no mês 1',
  'O maior valor médio de horas por visus do ano 2018 é 743.00 no mês 5',
  'O maior valor médio de horas por visus do ano 2019 é 743.00 no mês 3',
  'O maior valor médio de horas por visus do ano 2020 é 743.00 no mês 1',
  'O maior valor médio de horas por visus do ano 2021 é 743.00 no mês 8',
  'O maior valor médio de horas por visus do ano 2022 é 743.00 no mês 10',
  'O maior valor médio de horas por visus do ano 2023 é 743.00 no mês 1'],
 ['O menor valor médio de horas por visus do ano 2016 é 695.00 no mês 2',
  'O menor valor médio de horas por visus do ano 2017 é 671.00 no mês 2',
  'O menor valor médio de horas por visus do ano 2018 é 671.00 no mês 2',
  'O menor valor médio de horas por visus do ano 2019 é 671.00 no mês 2',
  'O menor valor médio de horas por visus do ano 2020 é 695.00 no mês 2',
  'O menor valor médio de horas por 

In [87]:
for num in range(2016, 2023):
    print(data[data['ano'] == num].groupby(['ano', 'mes', 'jogos_transmitidos'])['jogos_transmitidos'].sum().idxmax())

(2016, 12, 14033)
(2017, 12, 16871)
(2018, 7, 19290)
(2019, 5, 20751)
(2020, 12, 32840)
(2021, 5, 36960)
(2022, 7, 43566)


In [91]:
for num in range(2016, 2023):
    print(data[data['ano'] == num].groupby(['ano', 'mes', 'horas_assistidas'])['horas_assistidas'].sum().idxmax())

(2016, 3, 490669308)
(2017, 1, 601251200)
(2018, 11, 865630693)
(2019, 8, 999177398)
(2020, 12, 1877319317)
(2021, 5, 2308295297)
(2022, 1, 2162446961)


In [92]:
for num in range(2016, 2023):
    print(data[data['ano'] == num].groupby(['ano', 'mes', 'media_visualizacoes'])['media_visualizacoes'].sum().idxmax())

(2016, 3, 660389)
(2017, 1, 809220)
(2018, 11, 1203936)
(2019, 8, 1344787)
(2020, 12, 2526674)
(2021, 4, 3112681)
(2022, 2, 2956001)


<b><i>Provável que por conta da pandemia, o número de jogos e transmissões aumento muito a partir de 2020</i></b>

### Proporção de espectadores 

In [62]:
data.head()

ano  mes  horas_assistidas  media_visualizacoes  pico_de_visualizacoes  \
0  2016    1         480241904               646355                1275257   
1  2016    2         441859897               635769                1308032   
2  2016    3         490669308               660389                1591551   
3  2016    4         377975447               525696                1775120   
4  2016    5         449836631               605432                1438962   

   Streams  media_canais  jogos_transmitidos  proporcao_de_espectadores  
0  7701675         20076               12149                      29.08  
1  7038520         20427               12134                      28.98  
2  7390957         20271               12234                      28.92  
3  6869719         16791               12282                      28.80  
4  7535519         19394               12424                      28.85

In [69]:
data[data['ano'] == 2016].groupby(['ano', 'mes', 'proporcao_de_espectadores'])['proporcao_de_espectadores'].sum().idxmax()

(2016, 1, 29.08)

In [70]:
data[data['ano'] == 2017].groupby(['ano', 'mes', 'proporcao_de_espectadores'])['proporcao_de_espectadores'].sum().idxmax()

(2017, 1, 28.07)

In [71]:
data[data['ano'] == 2018].groupby(['ano', 'mes', 'proporcao_de_espectadores'])['proporcao_de_espectadores'].sum().idxmax()

(2018, 1, 26.02)

In [72]:
data[data['ano'] == 2019].groupby(['ano', 'mes', 'proporcao_de_espectadores'])['proporcao_de_espectadores'].sum().idxmax()

(2019, 8, 28.26)

In [73]:
data[data['ano'] == 2020].groupby(['ano', 'mes', 'proporcao_de_espectadores'])['proporcao_de_espectadores'].sum().idxmax()

(2020, 4, 26.79)

In [74]:
data[data['ano'] == 2021].groupby(['ano', 'mes', 'proporcao_de_espectadores'])['proporcao_de_espectadores'].sum().idxmax()

(2021, 10, 30.39)

In [75]:
data[data['ano'] == 2022].groupby(['ano', 'mes', 'proporcao_de_espectadores'])['proporcao_de_espectadores'].sum().idxmax()

(2022, 5, 29.4)

In [76]:
data[data['ano'] == 2023].groupby(['ano', 'mes', 'proporcao_de_espectadores'])['proporcao_de_espectadores'].sum().idxmax()

(2023, 1, 27.44)

In [92]:
pico = data.groupby(['Streams', 'proporcao_de_espectadores', 'media_canais'])['proporcao_de_espectadores'].sum().idxmax()
base = data.groupby(['Streams', 'proporcao_de_espectadores', 'media_canais'])['proporcao_de_espectadores'].sum().idxmin()

pico, base

((18209793, 30.39, 91858), (18859473, 22.65, 95781))

<b><i>O que se pode subentender dos dados acima é que não necessariamente o pico de espectadores está relacionado com os totais máximos de (1) canais, (2) transmissões e (3) números de streams ou até mesmo (4) com a média de visualizações

### Horas por Jogos

In [55]:
data[24:36]

ano  mes  horas_assistidas  media_visualizacoes  pico_de_visualizacoes  \
24  2018    1         720220030               969340                2315343   
25  2018    2         615710441               917601                1735587   
26  2018    3         721341613               970850                2399946   
27  2018    4         729326135              1014361                2009005   
28  2018    5         762177378              1025810                1714327   
29  2018    6         789065029              1097447                2847241   
30  2018    7         796242999              1071659                1974783   
31  2018    8         855860641              1151898                3929074   
32  2018    9         815061013              1133603                2431863   
33  2018   10         852724761              1147678                2093870   
34  2018   11         865630693              1203936                2515155   
35  2018   12         845969862              1138586                2021262   

     Streams  media_canais  jogos_transmitidos  proporcao_de_espectadores  
24  11222734         33580               18309                      26.02  
25  10729066         35548               18667                      25.82  
26  13495972         38981               19274                      25.79  
27  13101238         39491               19132                      25.80  
28  13869534         38746               19011                      25.76  
29  13888432         39761               18833                      25.62  
30  14910026         41629               19290                      25.41  
31  15769306         44383               19233                      25.34  
32  15005606         44246               18526                      25.23  
33  14934884         43558               18524                      25.11  
34  14422767         45464               18789                      24.76  
35  16271867         48190               19156                      24.36

In [52]:
data.groupby('ano')[['horas_assistidas', 'jogos_transmitidos']].max()

horas_assistidas  jogos_transmitidos
ano                                       
2016         490669308               14033
2017         601251200               16871
2018         865630693               19290
2019         999177398               20751
2020        1877319317               32840
2021        2308295297               36960
2022        2162446961               43566
2023        1866962462               43611

O que os dados acima nos permite entender é que o total máximo de horas assistidas não está relacionada ao total máximo de jogos transmitidos. 

### Empregando respostas

#### Maior e menor média de horas assistidas por visualização. Valores anuais e mensais. 

##### Valores anuais gerais

In [3]:
maior_horas = data.groupby(['ano', 'mes', 'horas_assistidas', 'media_visualizacoes'])['horas_assistidas'].sum().idxmax()
menor_horas = data.groupby(['ano', 'mes', 'horas_assistidas', 'media_visualizacoes'])['horas_assistidas'].sum().idxmin()

maior_horas, menor_horas

((2021, 5, 2308295297, 3106723), (2016, 9, 362833210, 504635))

In [2]:
print(f'A maior média de horas assistidas por visualizações foi de {maior_horas[2] / maior_horas[-1]:.2f}')
print(f'A menor média de horas assistidas por visualizações foi de {menor_horas[2] / menor_horas[-1]:.2f}')

A maior média de horas assistidas por visualizações foi de 743.00
A menor média de horas assistidas por visualizações foi de 719.00


Os trechos acima permitem compreender que, na data 05/2021 ocorreu o maior número bruto de horas assistidas 2308295297 com um total de 3106723 médias visualizações. Já o menor valor de horas assistidas ocorreu em 09/2016, com um número bruto de 362833210 horas assistidas e um total de 504635 média visualizações. <br>
A maior média horas / visus foi de 743 horas por média visualização, já o menor valor foi de 719 horas por média visualização.

##### Valores Mensais e anuais

In [5]:
def analisaMedias(horas:list(), visus:list()):
    aux = []
    total = []
    ano = 2016
    for x in range(85):
        if x < 12:
            aux.append(horas[x] / visus[x])
            if x == 11:
                aux.append(sum(horas[0:12]) / sum(visus[0:12]))
                aux.insert(0, ano)
                total.append(aux[:])
                aux.clear()
        elif 24 > x > 11:
            aux.append(horas[x] / visus[x])
            if x == 23:
                aux.append(sum(horas[11:24]) / sum(visus[11:24]))
                aux.insert(0, ano + 1)
                total.append(aux[:])
                aux.clear()
        elif 36 > x > 23:
            aux.append(horas[x] / visus[x])
            if x == 35:
                aux.append(sum(horas[24:36]) / sum(visus[24:36]))
                aux.insert(0, ano + 2)
                total.append(aux[:])
                aux.clear()
        elif 48 > x > 35:
            aux.append(horas[x] / visus[x])
            if x == 47:
                aux.append(sum(horas[36:48]) / sum(visus[36:48]))
                aux.insert(0, ano + 3)
                total.append(aux[:])
                aux.clear()
        elif 60 > x > 47:
            aux.append(horas[x] / visus[x])
            if x == 59:
                aux.append(sum(horas[48:60]) / sum(visus[48:60]))
                aux.insert(0, ano + 4)
                total.append(aux[:])
                aux.clear()         
        elif 72 > x > 59:
            aux.append(horas[x] / visus[x])
            if x == 71:
                aux.append(sum(horas[60:72]) / sum(visus[60:72]))
                aux.insert(0, ano + 5) 
                total.append(aux[:])
                aux.clear()
        elif 84 > x > 71:
            aux.append(horas[x] / visus[x])
            if x == 83:
                aux.append(sum(horas[72:84]) / sum(visus[72:84]))
                aux.insert(0, ano + 6)
                total.append(aux[:])
                aux.clear()       
        else:
            aux.append(horas[x] / visus[x])
            if x == 84:
                aux.append((horas[-1]) / (visus[-1]))
                aux.insert(0, ano + 7)
                total.append(aux[:])
                aux.clear()
    return total

   
        

In [6]:
horas_assistidas = [ dados for dados in data['horas_assistidas']]
media_visualizacoes = [dados for dados in data['media_visualizacoes']]

total_medias = analisaMedias(horas_assistidas, media_visualizacoes)

A variável <b>total_medias</b> corresponde a uma lista com outras 8 listas dentro de si. Cada lista dessas possui 1 determinado ano como valor inicial, 12 ou (1) valores de médias mensais de horas assistidas por número de espectadores e, por fim, um valor de média anual do total de horas assistidas pelo total de espectadores anuais.

In [7]:
maiores = []
minimos = []

for x in total_medias:
    maiores.append(f'O maior valor médio de horas por visus do ano {x[0]} é {max(x[1:-1]):.2f} no mês {x.index(max(x[1:-1]))}')
    minimos.append(f'O menor valor médio de horas por visus do ano {x[0]} é {min(x[1:-1]):.2f} no mês {x.index(min(x[1:-1]))}')
    
maiores, minimos

(['O maior valor médio de horas por visus do ano 2016 é 743.00 no mês 8',
  'O maior valor médio de horas por visus do ano 2017 é 743.00 no mês 1',
  'O maior valor médio de horas por visus do ano 2018 é 743.00 no mês 5',
  'O maior valor médio de horas por visus do ano 2019 é 743.00 no mês 3',
  'O maior valor médio de horas por visus do ano 2020 é 743.00 no mês 1',
  'O maior valor médio de horas por visus do ano 2021 é 743.00 no mês 8',
  'O maior valor médio de horas por visus do ano 2022 é 743.00 no mês 10',
  'O maior valor médio de horas por visus do ano 2023 é 743.00 no mês 1'],
 ['O menor valor médio de horas por visus do ano 2016 é 695.00 no mês 2',
  'O menor valor médio de horas por visus do ano 2017 é 671.00 no mês 2',
  'O menor valor médio de horas por visus do ano 2018 é 671.00 no mês 2',
  'O menor valor médio de horas por visus do ano 2019 é 671.00 no mês 2',
  'O menor valor médio de horas por visus do ano 2020 é 695.00 no mês 2',
  'O menor valor médio de horas por 

In [9]:
for x in range(8):
    print(f'O valor de média HA/MV anual de {total_medias[x][0]} é de {total_medias[x][-1]:.2f}')

O valor de média HA/MV anual de 2016 é de 731.19
O valor de média HA/MV anual de 2017 é de 729.96
O valor de média HA/MV anual de 2018 é de 729.54
O valor de média HA/MV anual de 2019 é de 728.81
O valor de média HA/MV anual de 2020 é de 731.42
O valor de média HA/MV anual de 2021 é de 728.77
O valor de média HA/MV anual de 2022 é de 728.38
O valor de média HA/MV anual de 2023 é de 743.00


In [1]:
for value in total_medias:
    print(f'O ano é {value[0]}')
    for index, values in enumerate(value):
        if 13 > index >= 1:
            print(f'O valor médio mensal {index} é {values:.2f} HA/MV')
            

NameError: name 'total_medias' is not defined

<b><i> O que se pode perceber após analisar todo o panorama geral do dataset é que as médias tendem a se manter iguais por meses e até mesmos anos. A variação anual não é tão significativa (considerando que 2023 está com 11 meses faltantes). É como se os dados da Twitch fossem estáveis, com poucas variações em perídos de (1) campeonatos famosos, (2) estreias de jogos famosos e (3) período de pandemia.

#### Meses de máxima, mínima e possíveis recorrências

In [14]:
for num in range(2016, 2023):
    print(data[data['ano'] == num].groupby(['ano', 'mes', 'horas_assistidas'])['horas_assistidas'].sum().idxmax())

(2016, 3, 490669308)
(2017, 1, 601251200)
(2018, 11, 865630693)
(2019, 8, 999177398)
(2020, 12, 1877319317)
(2021, 5, 2308295297)
(2022, 1, 2162446961)


In [15]:
for num in range(2016, 2023):
    print(data[data['ano'] == num].groupby(['ano', 'mes', 'horas_assistidas'])['horas_assistidas'].sum().idxmin())

(2016, 9, 362833210)
(2017, 2, 500932308)
(2018, 2, 615710441)
(2019, 9, 878083331)
(2020, 2, 981752149)
(2021, 11, 1812117484)
(2022, 12, 1674092138)


<b><i>Não chega a existir uma recorrência de ano para ano. Os valores de máxima se dão em períodos de lançamentos de jogos e/ou campeonatos. Há a questão ainda de começo, meio e fim da pandemia para explicar uma grande alta de horas assistidas, porém, faltam dados para comprovar que a mudança foi de fato a pandemia.

#### Proporção de Espectadores 

In [20]:
for x in range(2016, 2023):
    print(f'O ano é {x}')
    print(data[data['ano'] == x].groupby(['media_canais', 'proporcao_de_espectadores'])['proporcao_de_espectadores'].idxmax())

O ano é 2016
media_canais  proporcao_de_espectadores
16592         28.47                         7
16791         28.80                         3
16880         28.39                         8
18030         28.62                         6
18818         28.76                         5
19250         28.33                         9
19394         28.85                         4
20076         29.08                         0
20271         28.92                         2
20427         28.98                         1
20715         28.22                        10
22291         28.13                        11
Name: proporcao_de_espectadores, dtype: int64
O ano é 2017
media_canais  proporcao_de_espectadores
22078         27.60                        15
22362         27.33                        16
23094         27.73                        14
23412         27.18                        17
24081         28.07                        12
24219         27.85                        13
24711         26.89 

In [35]:
pro = [dado for dado in data['proporcao_de_espectadores']]
canais = [canal for canal in data['media_canais']]

In [44]:
aux_pro = pro[0]
aux_canais = canais[0]

for x in range(84):
    if x == 0:
        print((pro[x] >= aux_pro), (canais[x] >= aux_canais))
    else:
        print((pro[x] > aux_pro), (canais[x] > aux_canais))
    aux_pro = pro[x]
    aux_canais = canais[x]

True True
False True
False False
False False
True True
False False
False False
False False
False True
False True
False True
False True
False True
False True
False False
False False
False True
False True
False True
False False
False True
False True
False True
False True
False True
False True
False True
True True
False False
False True
False True
False True
False False
False False
False True
False True
True True
True True
True False
True False
True False
True False
False True
True True
False False
True False
True True
False True
False True
True True
False True
True True
False True
True False
False True
True True
True False
True True
False True
True True
True True
False True
True False
True False
True False
False False
False False
True False
True False
True False
False False
True False
True True
True True
False False
True False
True False
False True
False True
True False
False False
False True
False False
True False


<b><i>A interpretação dos dados acima permite compreender que não necessariamente o aumento da variável media_canais proporciona um aumento na variável media_espectadores. A variável media_espectadores seguiu-se padrão tal qual horas_assistidas, tendo aumento nos perídoso de alta no geral (campeonatos, lançamentos de jogos e pandemia), porém, no restante, não houve tanta variação.</i></b>

#### Horas assistidas por Jogos Transmitidos

In [48]:
for x in range(2016, 2023):
    print(data[data['ano'] == x].groupby(['ano', 'mes', 'horas_assistidas'])['jogos_transmitidos'].sum())

ano   mes  horas_assistidas
2016  1    480241904           12149
      2    441859897           12134
      3    490669308           12234
      4    377975447           12282
      5    449836631           12424
      6    446429345           12374
      7    446331825           12961
      8    382117373           13693
      9    362833210           13484
      10   453035001           13973
      11   449688428           13968
      12   490484523           14033
Name: jogos_transmitidos, dtype: int64
ano   mes  horas_assistidas
2017  1    601251200           14841
      2    500932308           14737
      3    516399977           14794
      4    535431301           15007
      5    506927522           15276
      6    510832330           15393
      7    566899230           15962
      8    533450366           15898
      9    527591318           15804
      10   585177534           15847
      11   575551503           16532
      12   573137507           16871
Name: jogos_trans

In [50]:
horas = [horas for horas in data['horas_assistidas']]
jogos = [jogos for jogos in data['jogos_transmitidos']]

aux_h, aux_j = horas[0], jogos[0]

for x in range(83):
    if x == 0:
        print((horas[x] >= aux_h), (jogos[x] >= aux_j))
    else:
        print((horas[x] > aux_h), (jogos[x] > aux_j))
    aux_h, aux_j = horas[x], jogos[x]

True True
False False
True True
False True
True True
False False
False True
False True
False False
True True
False False
True True
True True
False False
True True
True True
False True
True True
True True
False False
False False
True True
False True
False True
True True
False True
True True
True False
True False
True False
True True
True False
False False
True False
True True
False True
True True
False False
True True
False True
True True
False False
False False
True True
False False
True True
False True
True True
True True
False False
True True
True True
False True
False False
False True
True True
False True
True True
True True
True True
True True
False True
True True
True True
True True
False False
False True
True True
False False
True True
False False
True True
True True
False False
False True
False True
True True
False False
True True
True False
False False
True True
False False


<b><i> Ao analisar os dados acima, pode-se inferir que os períodos em que ambas as variáveis sofrem alguma alta são os períodos de máxima de outros valores também. Ou seja, em épocas de campeonatos, lançamentos de jogos e pandemia. Fora isso, não há correlação aparente.</i></b>